docs/architecture/adr-009.md

```diff
@@ -0,0 +1,99 @@
+# ADR 009: Permissioned addresses in Cronos
+
+## Changelog
+* 2022-01-04: first draft
+
+## Context
+
+Some messages in cronos require permissions. For example changing the mapping to support new CRC20 auto-conversion contracts or disabling the bridge module in case of emergency. Right now, only the admin has the ability to use those messages.
+The admin is a single address defined in cronos space and can be modified through governance. It is usually a multi-sig address shared by multiple trusted party in order to achieve a higher level of security.
+
+While a single multi-sig admin address was originally implemented with simplicity in mind, realistically it is not practical to rely on a single address to perform all admin tasks.
+As those operations could become more frequent (updating the token mapping) or needs to be trigger by external systems (circuit breaker for gravity module), it would be more practical to define a granular permission system which restrict certain operations to only "some" known addresses.
+
+
+## Terminologies
+
+### Admin
+
+A special address defined in cronos space that can be modified through governance. Usually a multi-sig address.
+
+### Permissioned addresses
+
+An address in Cronos that is allowed to perform a type of operation (message).
+
+### Non-permissioned addresses
+
+An address in Cronos that carry no permissions. 
+
+
+## Decision
+
+While there is some work-in-progress from Cosmos-sdk team to design a circuit breaker module (https://github.com/cosmos/cosmos-sdk/issues/14226) that could potentially resolve our issue, a temporary solution is to implement our own admin system in cronos for certain message type.
+
+After the circuit breaker is made available, those solutions could be either:
+- completely remove (rollback) and be replaced by the circuit breaker
+- partially removed and be replaced by the circuit breaker
```

Comment:
for` MsgUpdateTokenMapping` it is not related to circuit breaker. We might want to keep this solution for this msg and use the circuit breaker for `MsgTurnBridge`